src/sidebar/services/test/load-annotations-test.js

```diff
@@ -0,0 +1,268 @@
+import EventEmitter from 'tiny-emitter';
+
+import loadAnnotationsService, { $imports } from '../load-annotations';
+
+let searchClients;
+let longRunningSearchClient = false;
+class FakeSearchClient extends EventEmitter {
+  constructor(searchFn, opts) {
+    super();
+
+    assert.ok(searchFn);
+    searchClients.push(this);
+    this.cancel = sinon.stub();
+    this.incremental = !!opts.incremental;
+
+    this.get = sinon.spy(query => {
+      assert.ok(query.uri);
+
+      for (let i = 0; i < query.uri.length; i++) {
+        const uri = query.uri[i];
+        this.emit('results', [{ id: uri + '123', group: '__world__' }]);
+        this.emit('results', [{ id: uri + '456', group: 'private-group' }]);
+      }
+      if (!longRunningSearchClient) {
+        this.emit('end');
+      }
+    });
+  }
+}
+
+describe('loadAnnotationsService', () => {
+  let fakeStore;
+  let fakeApi;
+  let fakeAnnotationMapper;
+  let fakeStreamer;
+  let fakeStreamFilter;
+
+  let fakeUris;
+  let fakeGroupId;
+
+  beforeEach(() => {
+    sinon.stub(console, 'error');
+    searchClients = [];
+    longRunningSearchClient = false;
+
+    fakeAnnotationMapper = {
+      loadAnnotations: sinon.stub(),
+      unloadAnnotations: sinon.stub(),
+    };
+
+    fakeApi = {
+      search: sinon.stub(),
+    };
+
+    fakeStore = {
+      annotationFetchFinished: sinon.stub(),
+      annotationFetchStarted: sinon.stub(),
+      frames: sinon.stub(),
+      savedAnnotations: sinon.stub(),
+      updateFrameAnnotationFetchStatus: sinon.stub(),
+    };
+
+    fakeStreamer = {
+      setConfig: sinon.stub(),
+      connect: sinon.stub(),
+      reconnect: sinon.stub(),
+    };
+    fakeStreamFilter = {
+      resetFilter: sinon.stub().returns({
+        addClause: sinon.stub(),
+      }),
+      getFilter: sinon.stub().returns({}),
+    };
+    fakeUris = ['http://example.com'];
+    fakeGroupId = 'group-id';
+
+    $imports.$mock({
+      '../search-client': FakeSearchClient,
+    });
+  });
+
+  afterEach(() => {
+    console.error.restore();
+    $imports.$restore();
+  });
+
+  function service() {
```

Comment:
```suggestion
  function createService() {
```

It wasn't obvious when first reading the tests whether this function was creating a new instance each time or just returning some fixed instance.

Reply:
That's good feedback. This was legacy code from the previous incarnation of the annotations service, excuses excuses. I'll update.